src/implementation/Client/GRPCClient/GRPCClient.ts

```diff
@@ -112,7 +114,18 @@ export default class GRPCClient implements IClient {
     })
   }
 
+  async _startAwaitSidecarStarted(): Promise<void> {
+    await DaprClient.awaitSidecarStarted(async () => await GRPCClientSidecar.isStarted(this));
```

Comment:
Shall we add this as part of the `waitForReady` callback? So it will look something like:
```ts
return new Promise((resolve, reject) => {
  this.client.waitForReady(deadline, (err?) => {
        if (err) {
          this.logger.error(`Error waiting for client to be ready: ${err}`);
          reject();
        }
        await DaprClient.awaitSidecarStarted(async (err?) => {
            if (err) { reject(); }
            await GRPCClientSidecar.isStarted(this));
        }
        resolve();
    });
}
```

Reply:
If we do this, we can also remove `_startWaitForClientReady`, and directly configure the deadline and return the promise in the `start` method.

Reply:
They are now separate as 

```javascript
    await this._startAwaitSidecarStarted();
    await this._startWaitForClientReady();
```

One starts the gRPC connection, the other the sidecar, so I really want to keep them separate

Reply:
Got it, thanks for the clarification!